File: src/main/java/nl/devpieter/lobstar/models/player/PlayerStatus.java
package nl.devpieter.lobstar.models.player;

import nl.devpieter.lobstar.models.common.MinecraftVersion;

import java.util.UUID;

public record PlayerStatus(
        String name,

        long ping,
        boolean isOnline,

        String address,
        int port,

        UUID currentServerId,
        MinecraftVersion version
) {
}

File: src/main/java/nl/devpieter/lobstar/models/player/PlayerStatus.kt
package nl.devpieter.lobstar.models.player;

import nl.devpieter.lobstar.models.common.MinecraftVersion
import java.util.*

data class PlayerStatus(
    val name: String,

    val ping: Long,
    val isOnline: Boolean,

    val address: String,
    val port: Int,

    val requestedVirtualHost: String?,
    val matchingVirtualHostId: UUID?,

    val currentServerId: UUID,
    val version: MinecraftVersion
)